src/internal/packager/images/pull_test.go

```diff
@@ -6,14 +6,83 @@ package images
 
 import (
 	"context"
+	"encoding/json"
+	"fmt"
 	"os"
 	"path/filepath"
 	"testing"
 
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
 	"github.com/stretchr/testify/require"
 	"github.com/zarf-dev/zarf/src/pkg/transform"
 )
 
+func TestCheckForIndex(t *testing.T) {
+	t.Parallel()
+	testData := []struct {
+		name        string
+		ref         string
+		file        string
+		expectedErr string
+	}{
+		{
+			name:        "index sha",
+			ref:         "ghcr.io/zarf-dev/zarf/agent:v0.32.6@sha256:05a82656df5466ce17c3e364c16792ae21ce68438bfe06eeab309d0520c16b48",
+			file:        "agent-index.json",
+			expectedErr: pullIndexShaErr,
+		},
+		{
+			name:        "docker manifest list",
+			ref:         "defenseunicorns/zarf-game@sha256:0b694ca1c33afae97b7471488e07968599f1d2470c629f76af67145ca64428af",
+			file:        "game-index.json",
+			expectedErr: pullIndexShaErr,
+		},
+		{
+			name:        "image manifest",
+			ref:         "ghcr.io/zarf-dev/zarf/agent:v0.32.6",
+			file:        "agent-manifest.json",
+			expectedErr: "",
+		},
+		{
+			name:        "image manifest sha'd",
+			ref:         "ghcr.io/zarf-dev/zarf/agent:v0.32.6@sha256:b3fabdc7d4ecd0f396016ef78da19002c39e3ace352ea0ae4baa2ce9d5958376",
+			file:        "agent-manifest.json",
+			expectedErr: "",
+		},
+	}
+
+	for _, tc := range testData {
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+			refInfo, err := transform.ParseImageRef(tc.ref)
+			require.NoError(t, err)
+			file := filepath.Join("testdata", tc.file)
+			manifest, err := os.ReadFile(file)
+			require.NoError(t, err)
+			var idx v1.IndexManifest
+			jsonErr := json.Unmarshal(manifest, &idx)
+			require.NoError(t, jsonErr)
+			desc := &remote.Descriptor{
+				Descriptor: v1.Descriptor{
+					MediaType: idx.MediaType,
+				},
+				Manifest: manifest,
+			}
+			err = checkForIndex(refInfo, desc)
+			if err != nil {
```

Comment:
You probably want to check if expectedErr is not empty because otherwise this test will always pass. If err is nil we then assert that it is not nil.

Reply:
Yeah definitely